pwnlib/util/proc.py

```diff
@@ -340,3 +340,593 @@ def wait_for_debugger(pid):
         while tracer(pid) is None:
             time.sleep(0.01)
         l.success()
+
+
+class MemoryMapFlags:
+    """Object with the information of the flags field of maps described in /proc/<pid>/maps
+
+        Arguments:
+            readable(bool): True if map is readable
+            writable(bool): True if map is writable
+            executable(bool): True if map can be executed
+            private(bool): True if map is not shared, therefore private
+
+        Example:
+
+            >>> m = MemoryMapFlags.from_str('rw-p')
+            >>> m.readable
+            True
+            >>> m.writable
+            True
+            >>> m.executable
+            False
+            >>> m.private
+            True
+            >>> m.shared
+            False
+            >>> str(m)
+            'rw-p'
+    """
+
+    def __init__(self, readable, writable, executable, private):
+        self.readable = readable
+        self.writable = writable
+        self.executable = executable
+        self.private = private
+
+    @property
+    def shared(self):
+        """True if map is shared
+
+        Examples:
+            >>> m = MemoryMapFlags.from_str('rw-p')
+            >>> m.shared
+            False
+
+        Returns:
+            bool
+        """
+        return not self.private
+
+    @classmethod
+    def from_str(cls, flags_str):
+        """Generates a MemoryMapFlags object from a flags string. Example: 'r-xp'
+
+        Arguments:
+            flags_str(str): Flag string with the format contained in /proc/pid/maps
+
+        Returns:
+            MemoryMapFlags: The object representation of the string
+
+         Example:
+
+            >>> str(MemoryMapFlags.from_str('r-xp'))
+            'r-xp'
+        """
+        readable = flags_str[0] == "r"
+        writable = flags_str[1] == "w"
+        executable = flags_str[2] == "x"
+        private = flags_str[3] == "p"
+        return cls(readable, writable, executable, private)
+
+    def __str__(self):
+        flags_str = ""
+        flags_str += "r" if self.readable else "-"
+        flags_str += "w" if self.writable else "-"
+        flags_str += "x" if self.executable else "-"
+        flags_str += "p" if self.private else "s"
+        return flags_str
+
+
+class MemoryMap:
```

Comment:
Use `corefile.mappings`, this is unnecessary.

Reply:
Done in 0ca2072